conmon-rs/server/src/child_reaper.rs

```diff
@@ -0,0 +1,124 @@
+//! Child process reaping and management.
+
+use anyhow::{bail, Result};
+use getset::Getters;
+use log::{debug, error};
+use nix::sys::{
+    signal::{kill, Signal, SIGCHLD},
+    signalfd::{siginfo, SfdFlags, SigSet, SignalFd},
+};
+use nix::unistd::Pid;
+use std::path::PathBuf;
+use std::thread;
+use std::{collections::HashMap, fs::File, io::Write, sync::Arc, sync::Mutex};
+
+impl ChildReaper {
+    pub fn init_reaper(&self) -> Result<()> {
+        let mut mask = SigSet::empty();
+        mask.add(SIGCHLD);
+        mask.thread_block()?;
+        // Signals are queued up on the file descriptor
+        let mut sfd = SignalFd::with_flags(&mask, SfdFlags::empty())?;
+
+        let children = self.children.clone();
```

Comment:
This is a stylistic thing, but I prefer doing:
```suggestion
        let children = Arc::clone(self.children);
```

The rationale here is that `.clone()` can do almost anything - for example, you could be copying all of the backing storage behind a gigantic hash map.  In this case we're just bumping a reference count.  But to figure that out, I had to go look.  By using `Arc::clone()` we make that very explicit to the reader that this isn't actually an *expensive* clone.

Reply:
e.g. https://github.com/containers/containers-image-proxy-rs/blob/408dbded66f1bd2556cb697bb82f842eb4e401ae/src/imageproxy.rs#L281

Reply:
I like that